packages/web/src/components/navbar.tsx

```diff
@@ -0,0 +1,308 @@
+import { useNavigate, NavLink, Link } from 'react-router-dom';
+import {
+  PlusIcon,
+  ChevronDownIcon,
+  SparklesIcon,
+  ImportIcon,
+  TrashIcon,
+  ShareIcon,
+  ExternalLinkIcon,
+} from 'lucide-react';
+import { TitleCellType } from '@srcbook/shared';
+
+import { SessionType } from '@/types';
+import { SrcbookLogo } from '@/components/logos';
+import { Button } from '@/components/ui/button';
+import GenerateSrcbookModal from '@/components/generate-srcbook-modal';
+import DeleteSrcbookModal from '@/components/delete-srcbook-dialog';
+import { ExportSrcbookModal, ImportSrcbookModal } from '@/components/import-export-srcbook-modal';
+import useTheme from '@/components/use-theme';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuShortcut,
+  DropdownMenuTrigger,
+} from '@/components/ui/dropdown-menu';
+import { createSession, createSrcbook } from '@/lib/server';
+
+function LightDarkModeDebugChanger() {
+  const { theme, toggleTheme } = useTheme();
+
+  if (process.env.NODE_ENV === 'production') {
+    return null;
+  }
+
+  return (
+    <div className="absolute left-1/2 -translate-x-1/2">
+      <button
+        onClick={toggleTheme}
+        className="border-none outline-none text-muted-foreground hover:text-foreground font-semibold transition-colors"
+      >
+        {theme === 'light' ? '(DEV) Dark mode' : '(DEV) Light mode'}
+      </button>
+    </div>
+  );
+}
+
+type SessionNavbarProps = {
+  session: SessionType;
+  srcbooks: Array<SessionType>;
+  title: string;
+  baseDir: string;
+};
+
+export function SessionNavbar(props: SessionNavbarProps) {
+  const navigate = useNavigate();
+
+  const [showGenSrcbookModal, setShowGenSrcbookModal] = useState(false);
+  const [showImportSrcbookModal, setShowImportSrcbookModal] = useState(false);
+  const [showDelete, setShowDelete] = useState(false);
+  const [showSave, setShowSave] = useState(false);
+
+  async function openSrcbook(path: string) {
+    const { result: srcbook } = await createSession({ path });
+    navigate(`/srcbooks/${srcbook.id}`);
+  }
+
+  async function onCreateSrcbook() {
+    const { result } = await createSrcbook({
+      path: props.baseDir,
+      name: 'Untitled',
+
+      // FIXME: copy the same language as the currently visible srcbook - should this be less
+      // implicit?
+      language: props.session.language,
+    });
+    openSrcbook(result.path);
+  }
+
+  return (
+    <>
+      {/* FIXME: After closing this modal, `<body>` has `user-select: none;` on it? */}
+      <GenerateSrcbookModal
+        open={showGenSrcbookModal}
+        setOpen={setShowGenSrcbookModal}
+        openSrcbook={openSrcbook}
+      />
+      <ImportSrcbookModal open={showImportSrcbookModal} onOpenChange={setShowImportSrcbookModal} />
+      <DeleteSrcbookModal open={showDelete} onOpenChange={setShowDelete} session={props.session} />
+      <ExportSrcbookModal open={showSave} onOpenChange={setShowSave} session={props.session} />
```

Comment:
I have found that when any of these modals open and then close, the body has an inline `pointer-events: none;` style on it, and the app becomes somewhat non interactive:


https://github.com/user-attachments/assets/e083a0dd-0505-49ce-b9f9-b4963f1475ac

I need to look into this again and see what's going on. This doesn't happen on `main` so I think I introduced this inadvertently as part of my change.

Reply:
Figured this out, fixed in https://github.com/srcbookdev/srcbook/pull/267/commits/835e7f4128b9de34e6a91dec004cddff40aee776